### PR TITLE
feat: single use code request endpoint

### PR DESCRIPTION
**Pull request by YazeedLoonat**
## Issue Overview

This PR addresses [#issue](https://github.com/bloom-housing/bloom/issues/3899)

- [x] This change addresses the issue in full

## Description
This creates a new endpoint that allows a user to request a single use code

## How Can This Be Tested/Reviewed?
Tests were added, but the easiest way to test this is a bit complex. Follow these steps:

- Update the `jurisdictions` table to set `allow_single_use_code_login` to be true
- create a user via the public site
- modify the public site's sign in code to also call the new `authService.requestSingleUseCode` in addition to logging in 
- go to log in (you'll probably get an error if you didn't confirm your user). You should have received an email that matches the mock https://www.figma.com/file/2LSD4B4yBIxAUGbZQCVOCv?node-id=2901:7177&mode=design#716447984

## Checklist:

- [x] My code follows the style guidelines of this project
- [ ] I have added QA notes to the issue with applicable URLs
- [x] I have performed a self-review of my own code
- [x] I have reviewed the changes in a desktop view
- [ ] I have reviewed the changes in a mobile view
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have assigned reviewers
- [x] I have run `yarn generate:client` and/or created a migration if I made backend changes that require them
- [x] My commit message(s) is/are polished, and any breaking changes are indicated in the message and are well-described
- [x] Commits made across packages purposefully have the same commit message/version change, else are separated into different commits

## Reviewer Notes:

Steps to review a PR:

- Read and understand the issue, and ensure the author has added QA notes
- Review the code itself from a style point of view
- Pull the changes down locally and test that the acceptance criteria is met
- Also review the acceptance criteria on the Netlify deploy preview (noting that these do not yet include any backend changes made in the PR)
- Either explicitly ask a clarifying question, request changes, or approve the PR if there are small remaining changes but the PR is otherwise good to go

## On Merge:

If you have one commit and message, squash. If you need each message to be applied, rebase and merge.

### CODE REVIEW

File: api/prisma/migrations/00_init/migration.sql [hunk 1]
```diff
@@ -1292,4 +1292,11 @@ ALTER TABLE
 ADD
       CONSTRAINT "FK_87b8888186ca9769c960e926870" FOREIGN KEY ("user_id") REFERENCES "user_accounts"("id") ON
 DELETE CASCADE ON
-UPDATE CASCADE;
\ No newline at end of file
+UPDATE CASCADE;
```
Comment (emilyjablonski) [inline api/prisma/migrations/00_init/migration.sql:1295]: Are we able to update this migration as it has already run?

File: api/src/services/email.service.ts [hunk 2]
```diff
@@ -317,6 +317,27 @@ export class EmailService {
...
+    const jurisdiction = await this.getJurisdiction(user.jurisdictions);
+    void (await this.loadTranslations(jurisdiction, user.language));
+    const emailFromAddress = await this.getEmailToSendFrom(
+      user.jurisdictions,
+      jurisdiction,
+    );
+    await this.send(
+      user.email,
+      emailFromAddress,
+      'Partners Portal account access token',
```
Comment (emilyjablonski) [inline api/src/services/email.service.ts:330]: Is this maybe copy pasta text?

File: api/prisma/migrations/00_init/migration.sql (hunk 1 above)
Comment (YazeedLoonat) [reply api/prisma/migrations/00_init/migration.sql:1295]: So the change here is to give us translations locally and on the core environment whenever we reseed (`yarn setup`) 

we already have those migrations in HBA prod, so we don't want to insert that there


so I added a generalized version of HBA's translations for us here. We can run just that part separately on the core environment for now and then whenever we reseed there we'd have translations again

File: api/src/services/email.service.ts (hunk 2 above)
Comment (YazeedLoonat) [reply api/src/services/email.service.ts]: update! Em asked for 2 different subjects depending on if the user is a new user or a user logging in

lemme know what you think!

**Review by emilyjablonski — APPROVED** [review]

**Pull request merged**